### PR TITLE
Refactor noticelists controller to use crud component

### PR DESCRIPTION
**Pull request by righel**
## Generic requirements in order to contribute to MISP:

* One Pull Request per fix/feature/change/...
* Keep the amount of commits per PR as small as possible: if for any reason, you need to fix your commit after the pull request, please squash the changes in one single commit (or tell us why not)
* Always make sure it is mergeable in the default branch (as of today 2020-05-05: branch 2.4)
* Please make sure Travis CI works on this request, or update the test cases if needed
* Any major changes adding a functionality should be disabled by default in the config

#### What does it do?
Refactoring of /noticelists/index and noticelists/view to use CrudComponent.

#### Questions

- [ ] Does it require a DB change?
- [ ] Are you using it in production?
- [ ] Does it require a change in the API (PyMISP for example)?

### CODE REVIEW

File: app/Controller/NoticelistsController.php [hunk 1]
```diff
@@ -102,29 +102,34 @@ public function update()
...
-            $currentState['Noticelist']['enabled'] = 0;
-            $message = 'disabled';
-        } else {
-            $currentState['Noticelist']['enabled'] = 1;
-            $message = 'enabled';
-        }
-        if ($this->Noticelist->save($currentState)) {
-            return new CakeResponse(array('body' => json_encode(array('saved' => true, 'success' => 'Noticelist ' . $message)), 'status' => 200, 'type' => 'json'));
+        if ($this->request->is('post')) {
+            $this->Noticelist->read(null, $noticelist_id);
```
Comment (iglocska) [inline app/Controller/NoticelistsController.php]: This is generally not a good idea - find queries are preferred. The inheritence model of cake will actually fetch all notice list entries in this case

File: app/Controller/NoticelistsController.php [hunk 2]
```diff
@@ -112,29 +102,34 @@ public function update()
...
-            $currentState['Noticelist']['enabled'] = 1;
-            $message = 'enabled';
-        }
-        if ($this->Noticelist->save($currentState)) {
-            return new CakeResponse(array('body'=> json_encode(array('saved' => true, 'success' => 'Noticelist ' . $message)), 'status' => 200, 'type' => 'json'));
+        if ($this->request->is('post')) {
+            $this->Noticelist->read(null, $noticelist_id);
+
+            if (!$this->Noticelist->exists()) {
+                $message = 'Invalid Noticelist.';
```
Comment (mokaddem) [inline app/Controller/NoticelistsController.php]: Enable translation:
```php
$message = __('Invalid Noticelist.');
```

File: app/Controller/NoticelistsController.php [hunk 3]
```diff
@@ -200,4 +188,25 @@ public function delete($id)
...
         }
     }
+
+    public function preview_entries($id)
+    {
+        $this->set('menuData', ['menuList' => 'sync', 'menuItem' => 'previewNoticelistEntries']);
+
+        $noticelist = $this->Noticelist->find('first', array('contain' => array('NoticelistEntry'), 'conditions' => array('id' => $id)));
+        if (empty($noticelist)) {
+            throw new NotFoundException('Noticelist not found.');
```
Comment (mokaddem) [inline app/Controller/NoticelistsController.php]:
```php
throw new NotFoundException(__('Noticelist not found.'));
```

File: app/View/Noticelists/index.ctp [hunk 4]
```diff
@@ -1,85 +1,67 @@
...
-        ?>
-        </ul>
-    </div>
-</div>
 <?php
-    echo $this->element('/genericElements/SideMenu/side_menu', array('menuList' => 'noticelist', 'menuItem' => 'index'));
-?>
+$fields = [
+    [
+        'name' => __('Id'),
```
Comment (mokaddem) [inline app/View/Noticelists/index.ctp]:
```php
'name' => __('ID'),
```

File: app/View/Noticelists/index.ctp [hunk 5]
```diff
@@ -1,85 +1,67 @@
...
+        'data_path' => 'Noticelist.version',
+    ],
+    [
+        'name' => __('enabled'),
+        'data_path' => 'Noticelist.enabled',
+        'element' => 'toggle',
+        'url' => '/noticelists/toggleEnable',
+        'url_params_data_paths' => ['Noticelist.id'],
+        // 'checkbox_class' => 'defaultRoleCheckbox',
+        // 'beforeHook' => "$('.defaultRoleCheckbox').prop('checked', false); $(this).prop('checked', true);",
```
Comment (mokaddem) [inline app/View/Noticelists/index.ctp]: Remove?

**Comment by mokaddem** [comment]
As you introduced an new endpoint:
https://github.com/MISP/MISP/blob/748e043843e0aac90535d765db81a54a06acb532/app/Controller/NoticelistsController.php#L192
You have to create an entry in [ACLComponent.php](https://github.com/MISP/MISP/blob/2.4/app/Controller/Component/ACLComponent.php)

**Pull request merged**